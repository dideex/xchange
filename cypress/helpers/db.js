const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
