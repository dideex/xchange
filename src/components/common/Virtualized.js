import React, {Component} from 'react'
import {withRouter} from 'react-router-dom'
import styled from 'react-emotion'
import {Table, Column, AutoSizer} from 'react-virtualized'
import {CopyToClipboard} from 'react-copy-to-clipboard'
import PropTypes from 'prop-types'

import {StatusIconColors, StatusTitles} from './Styles'
import Icons from './Icons'

const TableWrap = styled('div')`
  padding: 10px 15px;
  border-radius: 10px;
  background-color: #ffffff;
`

const Status = styled('span')`
  display: block;
  width: 32px;
  height: 32px;
  background: ${({color}) => color};
  border-radius: 50%;
`

// Virtualized component;
@withRouter
export class Virtualized extends Component {
  static propTypes = {
    parsedOrders: PropTypes.array.isRequired,
    endpoint: PropTypes.string.isRequired,
  }
  state = {CopyId: null}

  render() {
    const {parsedOrders, endpoint} = this.props
    return (
      <TableWrap>
        <AutoSizer disableHeight>
          {({width}) => (
            <Table
              width={width}
              height={330}
              headerHeight={20}
              rowHeight={60}
              rowCount={parsedOrders.length}
              rowGetter={({index}) => parsedOrders[index]}
              onRowClick={({rowData: {id}}) =>
                this.props.history.push(`/${endpoint}/${id}`)
              }
            >
              <Column
                label="№"
                dataKey="id"
                width={100}
                cellRenderer={({cellData}) => (
                  <CopyToClipboard
                    text={cellData}
                    onCopy={() => this.setState({CopyId: cellData})}
                  >
                    <span title={cellData} onClick={e => e.stopPropagation()}>
                      <Icons
                        id="copy"
                        style={{width: 23}}
                        className={this.state.CopyId === cellData ? 'copyied' : ''}
                      />
                    </span>
                  </CopyToClipboard>
                )}
              />
              <Column label="Дата" dataKey="created" width={100} />
              <Column
                label="Сумма перевода"
                dataKey="inputValue"
                width={300}
                flexGrow={1}
              />
              <Column
                label="Сумма получения"
                dataKey="outputValue"
                width={300}
                flexGrow={1}
              />
              <Column label="На номер" dataKey="toWallet" width={300} flexGrow={1} />
              <Column
                label="Статус"
                dataKey="paymentStatus"
                width={100}
                cellRenderer={({cellData}) => (
                  <Status
                    title={StatusTitles[cellData]}
                    color={StatusIconColors[cellData]}
                  />
                )}
              />
            </Table>
          )}
        </AutoSizer>
      </TableWrap>
    )
  }
}

export default Virtualized
