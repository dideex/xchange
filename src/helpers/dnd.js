import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
