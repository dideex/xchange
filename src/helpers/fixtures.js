export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}