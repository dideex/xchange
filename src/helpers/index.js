export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
