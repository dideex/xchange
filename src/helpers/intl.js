import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
