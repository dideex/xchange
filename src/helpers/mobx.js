import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
