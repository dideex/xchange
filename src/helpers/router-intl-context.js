import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
