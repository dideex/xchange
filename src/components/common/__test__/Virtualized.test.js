import React from 'react'
import {Virtualized} from '../Virtualized'
import {mount, shallow} from 'enzyme'
import {StaticRouter} from 'react-router-dom'
import {mountWithIntl} from '../../../helpers/intl'

describe('Virtualized tests', () => {
  const intl = {formatMessage: ({id}) => id}
  it('Base markup', () => {
    const wrapper = mountWithIntl(
      <StaticRouter location="/" context={{}}>
        <Virtualized intl={intl} parsedOrders={[]} endpoint="endpoint" />
      </StaticRouter>,
    )
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('Orders behaviour', () => {
    it('Orderds markup', () => {
      const orders = [
        {
          id: 'id',
          created: 'created',
          inputValue: 'inputValue',
          outputValue: 'outputValue',
          toWallet: 'toWallet',
          paymentStatus: 'paymentStatus',
        },
      ]
      const wrapper = mountWithIntl(
        <StaticRouter location="/" context={{}}>
          <Virtualized intl={intl} parsedOrders={orders} endpoint="endpoint" />
        </StaticRouter>,
      )
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('Orderds markup', () => {
      const orders = [
        {
          id: 'id',
          created: 'created',
          inputValue: 'inputValue',
          outputValue: 'outputValue',
          toWallet: 'toWallet',
          paymentStatus: 'paymentStatus',
        },
        {
          id: 'id2',
          created: 'created2',
          inputValue: 'inputValue2',
          outputValue: 'outputValue2',
          toWallet: 'toWallet2',
          paymentStatus: 'paymentStatus2',
        },
        {
          id: 'id3',
          created: 'created3',
          inputValue: 'inputValue3',
          outputValue: 'outputValue3',
          toWallet: 'toWallet3',
          paymentStatus: 'paymentStatus3',
        },
      ]
      const wrapper = mountWithIntl(
        <StaticRouter location="/" context={{}}>
          <Virtualized intl={intl} parsedOrders={orders} endpoint="endpoint" />
        </StaticRouter>,
      )
      expect(wrapper.html()).toMatchSnapshot()
      console.log(wrapper.html())
    })
  })
})
