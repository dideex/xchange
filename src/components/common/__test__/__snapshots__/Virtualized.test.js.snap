// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Virtualized tests Base markup 1`] = `"<div class=\\"css-10q1pfo\\"><div style=\\"overflow: visible; width: 0px;\\"><div aria-colcount=\\"6\\" aria-rowcount=\\"0\\" class=\\"ReactVirtualized__Table\\" role=\\"grid\\"><div class=\\"ReactVirtualized__Table__headerRow\\" role=\\"row\\" style=\\"height: 20px; overflow: hidden; padding-right: 0px; width: 0px;\\"><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"№\\">№</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Date\\">Date</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Transfer amount\\">Transfer amount</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Recieve amount\\">Recieve amount</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"To wallet\\">To wallet</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Status:\\">Status:</span></div></div><div aria-label=\\"grid\\" aria-readonly=\\"true\\" class=\\"ReactVirtualized__Grid ReactVirtualized__Table__Grid\\" role=\\"rowgroup\\" style=\\"box-sizing: border-box; direction: ltr; height: 310px; position: relative; width: 0px; overflow-x: hidden; overflow-y: hidden;\\" tabindex=\\"0\\"></div></div></div><div class=\\"resize-triggers\\"><div class=\\"expand-trigger\\"><div style=\\"width: 1px; height: 1px;\\"></div></div><div class=\\"contract-trigger\\"></div></div></div>"`;

exports[`Virtualized tests Orders behaviour Orderds markup 1`] = `"<div class=\\"css-10q1pfo\\"><div style=\\"overflow: visible; width: 0px;\\"><div aria-colcount=\\"6\\" aria-rowcount=\\"1\\" class=\\"ReactVirtualized__Table\\" role=\\"grid\\"><div class=\\"ReactVirtualized__Table__headerRow\\" role=\\"row\\" style=\\"height: 20px; overflow: hidden; padding-right: 0px; width: 0px;\\"><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"№\\">№</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Date\\">Date</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Transfer amount\\">Transfer amount</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Recieve amount\\">Recieve amount</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"To wallet\\">To wallet</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Status:\\">Status:</span></div></div><div aria-label=\\"grid\\" aria-readonly=\\"true\\" class=\\"ReactVirtualized__Grid ReactVirtualized__Table__Grid\\" role=\\"rowgroup\\" style=\\"box-sizing: border-box; direction: ltr; height: 310px; position: relative; width: 0px; overflow-x: hidden; overflow-y: hidden;\\" tabindex=\\"0\\"></div></div></div><div class=\\"resize-triggers\\"><div class=\\"expand-trigger\\"><div style=\\"width: 1px; height: 1px;\\"></div></div><div class=\\"contract-trigger\\"></div></div></div>"`;

exports[`Virtualized tests Orders behaviour Orderds markup 2`] = `"<div class=\\"css-10q1pfo\\"><div style=\\"overflow: visible; width: 0px;\\"><div aria-colcount=\\"6\\" aria-rowcount=\\"3\\" class=\\"ReactVirtualized__Table\\" role=\\"grid\\"><div class=\\"ReactVirtualized__Table__headerRow\\" role=\\"row\\" style=\\"height: 20px; overflow: hidden; padding-right: 0px; width: 0px;\\"><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"№\\">№</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Date\\">Date</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Transfer amount\\">Transfer amount</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Recieve amount\\">Recieve amount</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"To wallet\\">To wallet</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Status:\\">Status:</span></div></div><div aria-label=\\"grid\\" aria-readonly=\\"true\\" class=\\"ReactVirtualized__Grid ReactVirtualized__Table__Grid\\" role=\\"rowgroup\\" style=\\"box-sizing: border-box; direction: ltr; height: 310px; position: relative; width: 0px; overflow-x: hidden; overflow-y: hidden;\\" tabindex=\\"0\\"></div></div></div><div class=\\"resize-triggers\\"><div class=\\"expand-trigger\\"><div style=\\"width: 1px; height: 1px;\\"></div></div><div class=\\"contract-trigger\\"></div></div></div>"`;
